### ISSUE TITLE
GraphQL Error: Unknown field `context` on type `SitePage`

### ISSUE THREAD

**Issue by mosanger** [gatsbyjs/gatsby#1837]
happening both in dev and production version, when context is queried in a layout.
Is this supposed to happen that context is not part of the SitePage query object?

Because if I edit the query and save the file everything seems to work after a reload. And the queries and layouts work...

layouts/main.jsx
```jsx
import React from 'react';
import PropTypes from 'prop-types';
import Layout from 'src/components/Layout';
import Footer from 'src/components/Footer';

const MainLayout = ({ children, ...rest }) => (
    <Layout {...rest}>
        <header>
            <Navigation />
        </header>
        <main>
             {children()}
        </main>
        <Footer />
    </Layout>
);

MainLayout.displayName = 'MainLayout';

MainLayout.propTypes = {
    children: PropTypes.func.isRequired,
};

export default MainLayout;

export const pageQuery = graphql`
    query MainLayoutQuery {
        site {
            siteMetadata {
                title
            }
        }
        allSitePage {
            edges {
                node {
                    path
                    context {
                        title
                    }
                }
            }
        }
    }
`;
```

**Comment by KyleAMathews** [comment]
The page context is already passed as the `pageContext` prop so you don't need to query it. It was a mistake adding it to graphql at all. I'll be removing it soon.

**Comment by mosanger** [comment]
I think I didn't state the issue very clearly...
I need the context of all the pages for menu building purposes, hence the `allSitePage` query.

**Comment by KyleAMathews** [comment]
Just query the original source of the data. We can't cleanly expose the context you add to pages like this.

**Comment by mosanger** [comment]
then your examples are a bit misleading... when this is not a reliable source of information.
I have a mix of markdown and js pages. and I'm using gatsby-node to consolidate info into the page context.

how else would I go about this? seems crazy to query all this in the layout doesn't it? :/

also... it does work after rebuild.

**Comment by KyleAMathews** [comment]
Which example?

**Comment by KyleAMathews** [comment]
There might be a way to set things up so querying pages makes sense but right now it'll break on you. If you're interested in adding better support, open and issue and we can discuss options.

**Comment by mosanger** [comment]
I've leaned heavily on the using-remark example, which is using context to transfer frontmatter data to page data. So I've used the same to combine remark and js frontmatter into my pages objects.

The only other way I found is using fields, but those are node-related, not page-related, which makes it a second choice option in my case. And as a wild guess, it's probably gonna be the same issue with fields regarding to queries in layouts.

Is there a possibility to make full queries available for layouts at startup? If so, I could open a pull request at some point...
